### PR TITLE
refactor: refactor ps filter to unify a common filter package

### LINKED ISSUE #2405: support a common filter for container, volume and images choose
## Why you need it?

```
pouch ps --filter
pouch images --filter
pouch volume --filter
```

pouch ps has implements a simple filter package, and pouch images and pouch volume are not support filter command, should get a common filter package for these command in daemon, and support the command in cli side. Optimize on current fileter package or implement a better one is wlecome.

### PR DESCRIPTION
**Pull request by ZYecho**
Signed-off-by: zhangyue <zy675793960@yeah.net>

### Ⅰ. Describe what this PR did
As the title desc.
refactor the test code and ps filter.
original pr #1595

### Ⅱ. Does this pull request fix one issue?

fix #2405 

### Ⅲ. Why don't you add test cases (unit test/integration test)? (你真的觉得不需要加测试吗？)
just refactor, wait for CI pass.

### Ⅳ. Describe how to verify it
wait for CI.

### Ⅴ. Special notes for reviews
support filter such as `label=key!=value`

### CODE REVIEW

File: apis/swagger.yml [hunk 1]
```diff
@@ -4159,33 +4159,6 @@ definitions:
       Link:
         type: "boolean"
 
-  ContainerListOptions:
```
Comment (allencloud) [inline apis/swagger.yml]: Good catch! Actually we have used this struct. Also described in https://github.com/alibaba/pouch/issues/2400.

File: apis/swagger.yml [hunk 2]
```diff
@@ -517,7 +517,7 @@ paths:
         - name: "filters"
           in: "query"
           description: |
-            Filters encoded as JSON string(type map[string][]string in Golang). This API will list containers match all of the filters. For example, `{"status": ["paused"]}` will only return paused containers.
+            Filters encoded as JSON string(type map[string]map[string]bool in Golang). This API will list containers match all of the filters. For example, `{"status": ["paused"]}` will only return paused containers.
```
Comment (allencloud) [inline apis/swagger.yml]: I am wondering if we have supported all these filter options. We should try to make the swagger definition and code logic consistent. cc @Ace-Tang for check.

For the un-implemented filter option or option in plan, we can mark TODO to tell readers.

File: apis/swagger.yml (hunk 1 above)
Comment (ZYecho) [reply apis/swagger.yml]: Maybe the not used warning was caused by not use in swagger.yaml, such as `$ref: '#/definitions/'`?
In addition, it's a question put this kind of struct into swagger or just define in the go file.

**Comment by ZYecho** [comment]
@Ace-Tang  PTAL

**Comment by Novicei** [comment]
Hello, if you currently want to merge this branch, you need to deal with the conflict.

**Pull request closed without merging**